src/index.ts

```diff
@@ -42,10 +45,16 @@ function makeBase58CheckEncoder(p2pkhVersion: number, p2shVersion: number): (dat
 function makeBase58CheckDecoder(p2pkhVersions: number[], p2shVersions: number[]): (data: string) => Buffer {
   return (data: string) => {
     const addr = bs58check.decode(data);
-    const version = addr.readUInt8(0);
-    if (p2pkhVersions.includes(version)) {
+    const version = addr.length === 21 ? addr.slice(0, 1) : addr.slice(0, 2);
```

Comment:
I think you need to modify this to take a list of versions, each of which is either a buffer or an array of bytes, and check for the version inside that array.

Also, is there some general-purpose way to determine how many version bytes there are, other than just checking the  string length?